lib/Parser.js

```diff
@@ -428,32 +428,36 @@ function parseFetch(text, literals, seqno) {
   var list = parseExpr(text, literals)[0], attrs = {}, m, body;
   // list is [KEY1, VAL1, KEY2, VAL2, .... KEYn, VALn]
   for (var i = 0, len = list.length, key, val; i < len; i += 2) {
-    key = list[i].toLowerCase();
-    val = list[i + 1];
-    if (key === 'envelope')
-      val = parseFetchEnvelope(val);
-    else if (key === 'internaldate')
-      val = new Date(val);
-    else if (key === 'modseq') // always a list of one value
-      val = ''+val[0];
-    else if (key === 'body' || key === 'bodystructure')
-      val = parseBodyStructure(val);
-    else if (m = RE_BODYINLINEKEY.exec(list[i])) {
-      // a body was sent as a non-literal
-      val = new Buffer(''+val);
-      body = new ReadableStream();
-      body._readableState.sync = false;
-      body._read = EMPTY_READCB;
-      this.emit('body', body, {
-        seqno: seqno,
-        which: m[1],
-        size: val.length
-      });
-      body.push(val);
-      body.push(null);
-      continue;
+    try {
+      key = list[i].toLowerCase();
```

Comment:
this is the line that currently errors, but figured I should wrap the whole method in try/catch.